mindsdb/api/http/namespaces/agents.py

```diff
@@ -410,6 +409,17 @@ def post(self, project_name, agent_name):
         # Trace Agent completions using Langfuse if configured.
         if os.getenv('LANGFUSE_PUBLIC_KEY') is not None:
 
+            # todo we need to fix this as this assumes that the model is always langchain
+            # since decoupling the model from langchain, we need to find a way to get the model name
+            # this breaks retrieval agents
+
+            # get model details
+            session = SessionController()
+            model_name_no_version, version = db.Predictor.get_name_and_version(existing_agent.model_name)
+            agent_model = session.model_controller.get_model(model_name_no_version, version=version,
+                                                             project_name=project_name)  # noqa
+            model_using = agent_model.get('problem_definition', {}).get('using', {})
```

Comment:
after decupling agent from langchain handler model could be not only mindsdb langchain model but:
- other type of llm model
- name of model for other providers (for example 'gpt-4' for openai or 'claude-2' for anthropic) 

In the future is better to move getting this info somewhere in (or after calling) [this function](https://github.com/mindsdb/mindsdb/blob/main/mindsdb/interfaces/agents/agents_controller.py#L40)

Reply:
yes see todo above that, this was added by a different PR. @paxcema is working on a fix for it here https://linear.app/mindsdb/issue/ML-105/fix-retrieval-of-model-details-for-tracing